### PR TITLE
Update spec to reflect that deleteCatalog is not cascading

### LINKED ISSUE #112: [BUG] delete catalog fails when there are existing tables and namespaces.
### Is this a possible security vulnerability?

- [X] This is NOT a possible security vulnerability

### Describe the bug

This is the description of delete catalog:
```
Delete an existing catalog. This is a cascading operation that deletes all metadata, including principals, roles and grants. If the catalog is an internal catalog, all tables and namespaces are dropped without purge.
```

However, when there are existing namespaces and tables, the request fails (instead of cascading).

### To Reproduce

1. Create a catalog.
2. Add namespaces and tables.
3. Try to delete the catalog.

### Actual Behavior

Request fails:

```
{
    "error": {
        "message": "Catalog 'zfab-test-1' cannot be dropped, it is not empty",
        "type": "BadRequestException",
        "code": 400
    }
}
```

### Expected Behavior

The catalog should be deleted including all namespaces and tables. (without purge).

### Additional context

_No response_

### System information

N/A

### PR DESCRIPTION
**Pull request by eric-maynard**
# Description

The language of the spec needs to be updated in order to accurately reflect the current behavior.

Partially fixes #112 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Doc change

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by flyrain — APPROVED** [review]
+1. Are we going to provide a tool for cascading deletion?

File: spec/index.yaml [hunk 1]
```diff
@@ -1209,11 +1209,7 @@ paths:
         - Polaris_Management_Service_OAuth2: []
     delete:
       operationId: deleteCatalog
-      description: >-
-        Delete an existing catalog. This is a cascading operation that deletes
-        all metadata, including principals, roles and grants. If the catalog is
-        an internal catalog, all tables and namespaces are dropped without
-        purge.
+      description: Delete an existing catalog. The catalog must be empty.
```
Comment (dimas-b) [inline spec/index.yaml:1212]: This PR itself LGTM, but I wonder how this spec file is related to the Iceberg REST Catalog spec... I see some paths match the Iceberg REST paths... I'd assume this whole file is meant to be 1:1 with the Iceberg spec... WDYT?

**Review by dimas-b — APPROVED** [review]

File: spec/index.yaml (hunk 1 above)
Comment (flyrain) [reply spec/index.yaml:1212]: Yes, it should be exact the same for anything in Iceberg REST spec. I think this file also provides specs that only in Polaris, like catalog endpoint.

Comment (dimas-b) [reply spec/index.yaml:1212]: I also opened a thread for this in Zulip Chat.

Comment (collado-mike) [reply spec/index.yaml:1212]: This spec file is specific to the Polaris management APIs - they aren't part of the Iceberg catalog spec, nor should they be. There shouldn't be any correspondence to catalog APIs. The actual Iceberg catalog rest spec that we implement lives here: https://github.com/apache/polaris/blob/main/spec/rest-catalog-open-api.yaml

Comment (flyrain) [reply spec/index.yaml:1220]: We got three yaml file in the directory
1. polaris-management-service.yml
2. rest-catalog-open-api.yaml
3. index.yaml

The index.yaml is actually a generated file combined by 1 and 2. I think that's why the confusion came from. Should we remove the generated file in the repo?
```
docker run -p 8080:80 -v ${PWD}:/spec --rm docker.io/redocly/cli join spec/docs.yaml spec/polaris-management-service.yml spec/rest-catalog-open-api.yaml -o spec/index.yaml --prefix-components-with-info-prop title
```

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
Thanks @eric-maynard and @dimas-b and @flyrain and @collado-mike  for Review

**Review by flyrain — APPROVED** [review]

**Comment by flyrain** [comment]
Thanks @eric-maynard for the fix. Thanks @dimas-b, @RussellSpitzer for the review.

**Pull request merged**